lib/functions/compilation/kernel.sh

```diff
@@ -98,7 +99,12 @@ function kernel_prepare_build_and_package() {
 		#["INSTALL_HDR_PATH"]="${kernel_dest_install_dir}/libc_headers" # Used by `make headers_install` - disabled, only used for libc headers
 	)
 
-	build_targets+=(install modules_install) # headers_install disabled, only used for libc headers
+	[ -z "${SRC_LOADADDR}" ] || install_make_params_quoted+=( "${SRC_LOADADDR}" ) # For uImage
+	# @TODO: Only combining `install` and `modules_install` enable mixed-build and __build_one_by_one
+	# We should spilt the `build` and `install` into two make steps as the kernel required
```

Comment:
it does perform a lot faster when done together, though; -jX when X>1 allows it to install/compress image while building modules, etc.

Reply:
No, it can't. Combing `install` and `modules_install` will enable mixed-build and __build_one_by_one.

https://github.com/torvalds/linux/blob/fc89d7fb499b0162e081f434d45e8d1b47e82ece/Makefile#L340-L345
```
# install and modules_install need also be processed one by one
ifneq ($(filter install,$(MAKECMDGOALS)),)
        ifneq ($(filter modules_install,$(MAKECMDGOALS)),)
		mixed-build := 1
        endif
endif
```

You can try to run `make all install` on a clean tree, it will fail immediately due to `install` may run before `all`.

```
[🌱 ] Building kernel [ meson linux-meson-current: 'LOADADDR=0x00208000' 'INSTALL_DTBS_PATH=/root/armbian/.tmp/work-b01c1178-9ddd-42ae-9f15-8d5dcd2b1a07/tmp.kgvPCR7G89/dtbs' 'INSTALL_PATH=/root/armbian/.tmp/work-b01c1178-9ddd-42ae-9f15-8d5dcd2b1a07/tmp.kgvPCR7G89/image/boot' 'INSTALL_MOD_PATH=/root/armbian/.tmp/work-b01c1178-9ddd-42ae-9f15-8d5dcd2b1a07/tmp.kgvPCR7G89/modules' all install dtbs_install ]
[🔨 ]   Statistics zeroed
[🔨 ]     SYNC    include/config/auto.conf.cmd
[🔨 ]     INSTALL /root/armbian/.tmp/work-b01c1178-9ddd-42ae-9f15-8d5dcd2b1a07/tmp.kgvPCR7G89/image/boot
[🔨 ]   
[🔨 ]    *** Missing file: arch/arm/boot/Image
[🔨 ]    *** You need to run "make" before "make install".
[🔨 ]   
[🔨 ]   make: *** [arch/arm/Makefile:302: install] Error 1
```

Reply:
Funny, I thought that was exactly what we did. https://github.com/armbian/build/blob/main/lib/functions/compilation/kernel.sh#L92